### PR TITLE
Add timetagger

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Motivation for this change

Want to try this app

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

The `./result` is empty after a successful build, and I don't understand why (I am not a python guy) ...

Maybe some nixpkgs+python guru can tell me?

I also want to write a service module as soon as I can test this!

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@FRidh and @jonringer I am ready for review here - if you can teach me how to add a test for the service definition, I'd write one :laughing:

File: nixos/modules/services/web-apps/timetagger.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      };
+    };
+  };
+
+  config = let
+    timetaggerPkg = if !isNull cfg.package then cfg.package else
+      pkgs.timetagger.overwriteAttrs {
+        addr = cfg.bindAddr;
+        port = cfg.port;
+      };
```
Comment (aanderse) [inline nixos/modules/services/web-apps/timetagger.nix:58]: If you take my advice above with setting the `default` in `package` then this can be dropped.

**Review by aanderse — CHANGES_REQUESTED** [review]
I left some comments on the module portion of this PR. I hope you find my suggestions useful.

**Comment by matthiasbeyer** [comment]
@aanderse thanks for your suggestions. Please have a look at https://github.com/NixOS/nixpkgs/pull/151364/commits/1f7534201bf05868c270f1ccccfab45ade4f83d3 where I applied the change with the package default value.

I differed from your suggestion a little bit, most notably the `defaultText` option, where I used interpolation of `cfg.bindAddr` and `cfg.addr`, which I guess is more understandable if the user sees it in the nixos option search (in contrast to the strings `bindAddr` and `port`). What do you think about this part?

---

If you approve I will rebase and squash my fixup (the minor changes), but leave the commit mentioned above as standalone for better traceability.

**Comment by matthiasbeyer** [comment]
About the handling for port and bind addr: Maybe these should be options for the `run.py` script upstream provides. I think this cannot  (should not) be fixed in nixpkgs but upstream - With options (optionally with default values to not break backwards compatibility) for the `run.py` script.

**Comment by aanderse** [comment]
Looks good 👍 Yeah, I was a bit shocked that this software doesn't have a way to set these values from either the cli or a config file. Unusual.

File: nixos/modules/services/web-apps/timetagger.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  options = {
+    services.timetagger = {
+      enable = mkEnableOption ''
+        Tag your time, get the insight
+
+        Note: This app does not do authentication.
+        You must setup authentication yourself or run it in an environment where
+        only allowed users have access.
+      '';
```
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/web-apps/timetagger.nix]: Please see definition of `mkEnableOption` to better understand my suggestion here. If you render the NixOS manual and look at the generated options page this looks strange as is.

**Comment by matthiasbeyer** [comment]
Me too. Either way, squashed and pushed. Ready to merge from my POV. :smile:

**Comment by matthiasbeyer** [comment]
Tests!!! :partying_face: :tada: 
With help from @0x4A6F :heart:

**Comment by 0x4A6F** [comment]
- [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] (Release notes changes) Ran nixos/doc/manual/md-to-db.sh to update generated release notes

As a side note: We have `listenAddress`/`(listen.)address` and a variation of `listen`, `bind`, `ip` and `addr` occurrences in modules instead of `bindAddr`.

**Review by 0x4A6F — APPROVED** [review]

**Comment by aanderse** [comment]
Module looks fine to me :+1:

**Comment by 0x4A6F** [comment]
Thanks!

**Comment by 0x4A6F** [comment]
> Module looks fine to me +1

Module is not included in `nixos/modules/module-list.nix` and does need a test.

@matthiasbeyer Do you have an example running this service?

**Comment by matthiasbeyer** [comment]
No, I just tested it once without even using a service definition IIRC.

**Pull request merged**